File: in_Python/1290 Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getDecimalValue(self, head):
        """
        :type head: ListNode
        :rtype: int
        """
        temp = []
        while head != None:
            temp.insert(0,head.val)
            head = head.next
        res = 0
        for i in range(len(temp)):
            res += temp[i] * (2 ** i)
        return res
        

File: in_Java/1290 Convert Binary Number in a Linked List to Integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        int l = 0;
        ListNode d = head;
        while(d != null){
            d = d.next;
            l++;
        }
        int res = 0;
        while(head != null){
            res += head.val * (Math.pow(2,l-1));
            head = head.next;
            l--;
        }
        return res;
    }
}